src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const theLongestSide = Math.max(a, b, c);
+    const sumOfThoSmallest = a + b + c - theLongestSide;
+
+    if (a <= 0 || b <= 0 || c <= 0 || sumOfThoSmallest <= theLongestSide) {
```

Comment:
[CODE STYLE] - The error message thrown here could be more descriptive. Instead of saying 'Each side ought be >0. And the longest ought be less than others sum', consider explaining that the sides provided do not form a valid triangle.